woocommerce/woocommerce
# Add/34000 attribute select control

### All Submissions:






### Changes proposed in this Pull Request:

This PR enables the select2 typeahead field for the attribute dropdown and the terms dropdown within the individual product attributes.
Currently I added a threshold for the attribute dropdown of 20, if more it will show the async typeahead field. This is similar to what is done for the category dropdown in the product list page. I could remove this condition if we don't think this is necessary.

https://user-images.githubusercontent.com/2240960/191288330-704e81fa-7221-46a2-9fe8-1feb046b4e9e.mp4

Partially address #34000

### How to test the changes in this Pull Request:

1. Load and build this PR and finish the onboarding, adding several products and attributes (an easy way is to import the [sample products CSV](https://github.com/woocommerce/woocommerce/tree/trunk/plugins/woocommerce/sample-data)).
2. Either edit an existing product or create a new one and select the **Attributes** tab
3. Assume you still have attributes less then 20, the normal dropdown should show.
4. Add the color attribute and add some terms. Notice how the terms are loaded on the fly and filtered as you type.
5. Now either add more then 20 different attributes (would probably have to be done manually) or add this filter to your `mu-plugin` -> `add_filter( 'woocommerce_attribute_taxonomy_filter_threshold', function() { return 1; } )`
6. It should now show a **Add custom attribute** button and a typeahead dropdown in the Attributes tab.
7. Test out the **Add custom attribute** button, it should add the custom item.
8. Now search for existing attributes and upon selection it should add the attribute to the product
9. Try searching for your previously selected item, it should be disabled now.
10. Now remove the attribute, the dropdown should not show it as disabled anymore.
11. Saving the attributes and persisting by refreshing the page should work the same as usual.

### Other information:




### FOR PR REVIEWER ONLY:
